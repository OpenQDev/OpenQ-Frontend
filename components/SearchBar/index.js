// Third party
import React from 'react';

const SearchBar = ({ onKeyUp, placeholder, searchText, label, styles }) => {
  return (
    <input
      className={`flex-1 lg:col-span-3 col-span-4 input-field-big ${styles}`}
      onChange={(e) => onKeyUp(e)}
      value={searchText}
      aria-label={label}
      type='text'
      placeholder={placeholder}
    ></input>
  );
};

export default SearchBar;
