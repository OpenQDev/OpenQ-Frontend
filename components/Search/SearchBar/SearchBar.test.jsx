/**
 * @vi-environment jsdom
 */
import React from 'react';
import SearchBar from '.';
import renderer from 'react-test-renderer';

describe('SearchBar', () => {
  const onKeyUp = vi.fn();
  it('should match DOM Snapshot', () => {
    const tree = renderer.create(<SearchBar onKeyUp={onKeyUp} placeholder='test' searchText={''} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
