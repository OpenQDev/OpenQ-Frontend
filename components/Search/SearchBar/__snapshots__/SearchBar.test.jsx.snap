// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar > should match DOM Snapshot 1`] = `
<input
  className="flex-1 lg:col-span-3 col-span-4 input-field-big undefined"
  onChange={[Function]}
  placeholder="test"
  type="text"
  value=""
/>
`;
